### PR TITLE
feat: added the no-useless-children-snippet rule

### LINKED ISSUE #909: Add rule `no-useless-children-snippet`
### Motivation

Any content inside a component tag becomes the `children` snippet. Thus, If you have a component with a children snippet without parameters, that's unnecessary.

### Description

I'd like to add a rule that bans the `children` snippet if it has no parameters

### Examples

```svelte
<script>
</script>

<!-- ✓ GOOD -->
<Foo>
  {#snippet bar()}
    Hello
  {/snippet}
</Foo>

<Foo>
  {#snippet children(val)}
    Hello {val}
  {/snippet}
</Foo>

<Foo>
  Hello
</Foo>

<!-- ✗ BAD -->
<Foo>
  {#snippet children()}
    Hello
  {/snippet}
</Foo>
```

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by marekdedic**
Closes #909

### CODE REVIEW

**Comment by marekdedic** [comment]
Blocked by #923 which causes the CI errors

**Comment by ota-meshi** [comment]
Rule implementation looks good, but I made this package ESM.
https://github.com/sveltejs/eslint-plugin-svelte/pull/932

Could you please merge the main branch and fix it by adding `.js` to the import statement of your implementation?

**Review by baseballyama — COMMENTED** [review]
In cases like this code, “Found an unnecessary children snippet.” appears, but this is incorrect. It would be better to only inspect inside Svelte components.

```svelte
{#snippet children()}
  Hi!
{/snippet}

{@render children()}
```

https://svelte.dev/playground/hello-world?version=5.16.0#H4sIAAAAAAAAE23LywrCMBSE4Vep46aFovusdNd3sC5qM2IgnobkeCPk3SWgC8HtN_9kyHQlDAZ6vzSPJXrbtLROaTv0ODvPBHPI0FeoXQX039c-hE2602u105T4z-dFlKIJBnmdxIVAbeaL8zZS2q6MMurgVqPk7WetJHkXKZbxN0UP5VNhNN5YjuUNEZNJhcEAAAA=

**Comment by marekdedic** [comment]
@baseballyama Good catch, fixed & added it as a test :)

File: packages/eslint-plugin-svelte/src/rules/no-useless-children-snippet.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		messages: {
+			uselessSnippet: 'Found an unnecessary children snippet.'
+		},
+		type: 'suggestion'
+	},
+	create(context) {
+		return {
+			SvelteSnippetBlock(node) {
+				if (
+					node.parent.type === 'SvelteElement' &&
```
Comment (baseballyama) [inline packages/eslint-plugin-svelte/src/rules/no-useless-children-snippet.ts:20]: We need to check `<svelte:component>` also.

https://svelte.dev/playground/hello-world?version=5.16.0#H4sIAAAAAAAAE3VQwW7CMAz9FeMdAKmi96xDmnbhH-gO0BoRKbWjxLBNUf59SqGaitjNfn7v2X4J-TAQGtyRcwJfElwPK-qtUr_GCk_WUUSzT6g_vvAKgNWkevd-E6_ktGDHQ6RneCesxBrRYHMDjXi1whHChSlCvW255SZ2wXottdrBS1D4kMHDKcgAy01dGmFivTsvX1tu6j8RT-bdRAQ92_iWijCPtuklsvWeFLqzdX0gXq1zywAA4_-lTPWdk0f7B8stVqj0rWg0XChX_-TyeOs8nSfTWUZObins7GK8ofTzvZ_5Fwm0cgS5AQAA

**Review by baseballyama — APPROVED** [review]

**Pull request merged**